rust/lance/src/io/commit.rs

```diff
@@ -625,6 +626,7 @@ pub(crate) async fn do_commit_detached_transaction(
 
         match result {
             Ok(path) => {
+                auto_cleanup_hook(dataset, &manifest).await?;
```

Comment:
For detached versions, the `manifest.version` value is a random integer, with the highest bit set. So I think the `interval` check will be meaningless in this case. I think we can remove this and just keep this in the `commit_transaction` code path.